### PR TITLE
fix(DDS): fix dds database user

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

1. fix timeout
2. fix doc
3. remove param validation
4. add API comments
5. add retry to create, update and delete method

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix dds database user
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dds/ TESTARGS='-run TestAccDatabaseUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds/ -v -run TestAccDatabaseUser_basic -timeout 360m -parallel 4
=== RUN   TestAccDatabaseUser_basic
=== PAUSE TestAccDatabaseUser_basic
=== CONT  TestAccDatabaseUser_basic
--- PASS: TestAccDatabaseUser_basic (1084.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       1084.327s
```

### CODE REVIEW

File: docs/resources/dds_database_user.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ subcategory: "Document Database Service (DDS)"
 
 # huaweicloud_dds_database_user
 
-Manages a database user resource within HuaweiCloud.
+Manages DDS database user resource within HuaweiCloud.
```
Comment (Jason-Zhang9309) [inline docs/resources/dds_database_user.md]:
```suggestion
Manages a DDS database user resource within HuaweiCloud.
```

File: docs/resources/dds_database_user.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ The following arguments are supported:
 * `instance_id` - (Required, String, ForceNew) Specifies the DDS instance ID to which the user belongs.
   Changing this parameter will create a new user.
 
-* `name` - (Required, String, ForceNew) Specifies the user name.
+* `name` - (Required, String, ForceNew) Specifies the user-name.
```
Comment (Jason-Zhang9309) [inline docs/resources/dds_database_user.md]: use `user name` or `username`

Comment (houpeng80) [reply docs/resources/dds_database_user.md]: done

File: docs/resources/dds_database_user.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/dds_database_user.md]: done

**Pull request merged**